### PR TITLE
[Bug fix]add yolov7face landmarks

### PR DESCRIPTION
**Pull request by CoolKbh**
### Description
补充yolov7face预测图片的landmarks
solved: https://github.com/PaddlePaddle/FastDeploy/issues/1292
solved: https://github.com/PaddlePaddle/FastDeploy/issues/1281

### CODE REVIEW

File: python/fastdeploy/vision/facedet/contrib/yolov7face.py [hunk 1]
```diff
@@ -119,6 +126,11 @@ def nms_threshold(self, nms_threshold):
             "The value to set `nms_threshold` must be type of float."
         self._postprocessor.nms_threshold = nms_threshold
 
+    @landmarks_per_face.setter
+    def nms_threshold(self, landmarks_per_face):
```
Comment (wjj19950828) [inline python/fastdeploy/vision/facedet/contrib/yolov7face.py]: nms_threshold？

File: fastdeploy/vision/facedet/contrib/yolov7face/postprocessor.cc [hunk 2]
```diff
@@ -61,6 +63,15 @@ bool Yolov7FacePostprocessor::Run(const std::vector<FDTensor>& infer_result,
       (*results)[bs].boxes.emplace_back(std::array<float, 4>{
           (x - w / 2.f), (y - h / 2.f), (x + w / 2.f), (y + h / 2.f)});
       (*results)[bs].scores.push_back(confidence);
+
+      // decode landmarks (default 5 landmarks)
+      if (landmarks_per_face_ > 0) {
+        float* landmarks_ptr = const_cast<float*>(reg_cls_ptr + 6);
+        for (size_t j = 0; j < landmarks_per_face_ * 3; j += 3) {
+          (*results)[bs].landmarks.emplace_back(
+              std::array<float, 2>{landmarks_ptr[j], landmarks_ptr[j + 1]});
```
Comment (wjj19950828) [inline fastdeploy/vision/facedet/contrib/yolov7face/postprocessor.cc:72]: 这里为什么是每隔三个取一次，但是只用到了前两个？

**Review by wjj19950828 — CHANGES_REQUESTED** [review]

File: fastdeploy/vision/facedet/contrib/yolov7face/postprocessor.cc (hunk 2 above)
Comment (CoolKbh) [reply fastdeploy/vision/facedet/contrib/yolov7face/postprocessor.cc:72]: `
    def extract_boxes(self, predictions):
        # Extract boxes from predictions
        # print(predictions[:, :4])
        boxes = predictions[:, :4] #* self.scale
        kpts = predictions[:, 6:]  ###x1,y1,score1, ...., x5,y5,score5
        kpts *= np.tile(np.array([self.scale[0], self.scale[1], 1], dtype=np.float32), (1, 5))`
repo里写的是x,y,score

File: python/fastdeploy/vision/facedet/contrib/yolov7face.py [hunk 3]
```diff
@@ -119,6 +126,11 @@ def nms_threshold(self, nms_threshold):
             "The value to set `nms_threshold` must be type of float."
         self._postprocessor.nms_threshold = nms_threshold
 
+    @landmarks_per_face.setter
+    def landmarks_per_face(self, landmarks_per_face):
+        assert isinstance(landmarks_per_face, float),\
```
Comment (wjj19950828) [inline python/fastdeploy/vision/facedet/contrib/yolov7face.py]: log是int，但是判断是float

**Review by wjj19950828 — CHANGES_REQUESTED** [review]

File: python/fastdeploy/vision/facedet/contrib/yolov7face.py (hunk 3 above)
Comment (CoolKbh) [reply python/fastdeploy/vision/facedet/contrib/yolov7face.py]: done

**Review by wjj19950828 — APPROVED** [review]

**Review by jiangjiajun — APPROVED** [review]

**Pull request merged**